src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/pullbased/ResultElementIterWithBinaryExecOp.java

```diff
@@ -27,6 +29,18 @@ public BinaryExecutableOp getOp() {
 		return opRunnerThread.getOp();
 	}
 
+	@Override
+	public int getArity() {
+		return 2;
+	}
+
+	@Override
+	public ResultElementIterator getSubIterator(final int i) throws NoSuchElementException {
```

Comment:
Providing access to the input iterators is a bit dangerous because this opens the possibility that they are accidentally consumed elsewhere even if these iterators are meant to be consumed here.